megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -2455,8 +2455,8 @@ FiringPlan getBestFiringPlan(final Entity shooter,
      * @param useExtremeRange Is the extreme range optional rule in effect?
      * @return The most damage done at that range.
      */
-    // todo cluster and other variable damage.
-    double getMaxDamageAtRange(final Entity shooter,
+    // todo: cluster and other variable damage.
+    public static double getMaxDamageAtRange(final Entity shooter,
```

Comment:
This computation needs to be cached, as it is expensive and carried out repeatedly.